iceoryx_hoofs/README.md

```diff
@@ -39,112 +22,136 @@ and we do not use dynamic memory. In these cases we adjusted the API to our use
+|:---------------------:|:--------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|`filesystem`           |          | Implementation of C++17 filesystem features for instance `cxx::perms` to abstract file permissions                                                                                                                                    |
+|`AccessController`     | i        | Interface for Access Control Lists (ACL).                                                                                                                                                                                             |
+|`FileLock`             |          | File lock C++ wrapping class.                                                                                                                                                                                                         |
+|`posix_access_rights`  |          | Rights and user management interface.                                                                                                                                                                                                 |
+
+### Functional (functional)
+
+| class                 | internal | description                                                                                                                                                                                                                            |
+|:---------------------:|:--------:|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|`storable_function`    | i        | A `std::function` alternative with configurable backend for memory storage.                                                                                                                                                            |
+|`function`             |          | A stack-based `std::function` replacement based on `storable_function`                                                                                                                                                                 |
+|`function_ref`         |          | C++11 implementation of the next-gen C++ feature `std::function_ref` see [function_ref proposal](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0792r2.html). It behaves like `std::function` but does not own the callable. |
+
+### Utility (utility?)
```

Comment:
```suggestion
### Utility (utility)
```

Reply:
@FerdinandSpitzschnueffler The question mark was added intentionally :upside_down_face: It's a hint for our future selves to think about a better name for `Utility`. Although, the STL also defines a [Utility](https://en.cppreference.com/w/cpp/utility) library :confused: If we don't find anything better till `v3.0`, we'll stick with utility.

Reply:
@mossmaurice What about Pinöpel? Just kidding...